config/initializers/mime_types.rb

```diff
@@ -1,13 +1,5 @@
-# Be sure to restart your server when you modify this file.
-
-# Add new mime types for use in respond_to blocks:
-# Mime::Type.register "text/richtext", :rtf
-
 # for azw3 files
-Mime::Type.register 'application/x-mobi8-ebook', :azw3
```

Comment:
Ok, so this is where things get a bit confusing. Even though the calls here look the same as in the files above, the casing difference is vital. The former, `MIME`, is what we want to remove. The one here actually belongs to Rails's internal plumbing to figure out how to label request types properly: https://github.com/rails/rails/blob/44db6bbfcbe66ac5926f01f3b693a378552b752f/actionpack/lib/action_dispatch/http/mime_type.rb#L183

What was happening was:
* Rails gets a request for extension `mobi` and doesn't know what it is, so labels it with the null mimetype as per https://github.com/rails/rails/blob/44db6bbfcbe66ac5926f01f3b693a378552b752f/actionpack/lib/action_dispatch/http/mime_negotiation.rb#L64
* The respond_to block in https://github.com/otwcode/otwarchive/blob/34bfb3c642a1b304412315df837050e397b9f97c/app/controllers/downloads_controller.rb#L10 doesn't have something for the null type, so the controller action bails with a 404 HTML page instead of the download

From my testing in Docker, adding these back in (but probably keeping the `Marcel` ones too) should fix the tests. (On a related note, if you want to run the tests in docker locally you'll need to add `- QTWEBENGINE_CHROMIUM_FLAGS=--no-sandbox` to https://github.com/otwcode/otwarchive/blob/34bfb3c642a1b304412315df837050e397b9f97c/docker-compose.yml#L99)

Reply:
Thanks to you i'm finalllyyyy finishing with this hahaha, I've been unable to test locally due to an issue with the db, which apparently doesn't find the conf file in the path when running `docker-compose run --rm test bundle exec cucumber features`, but the normal `docker-compose up` does work, so idk, I'm working on fixing that to be able to test locally instead of doing multiple tries in the PRs